tests/integration/steps/integration_steps.go

```diff
@@ -190,52 +242,56 @@ func resourceWithLabelIsPresent(t *testing.T, namespace, resourceType, label str
-	clientset, _ := k8s.GetKubernetesClientFromOptionsE(t, getKubectlOptions(namespace))
-	services, err := clientset.CoreV1().Services(namespace).List(context.Background(), metav1.ListOptions{LabelSelector: label})
-	g(t).Expect(err).To(BeNil(), fmt.Sprintf("Failed listing services with label %s", label))
+func getServicesWithLabels(t *testing.T, namespace string, labels map[string]string) *v1.ServiceList {
+	services := &v1.ServiceList{}
+	err := testClient.List(context.Background(), services, client.InNamespace(namespace), client.MatchingLabels(labels))
+	g(t).Expect(err).To(BeNil(), fmt.Sprintf("Failed listing services with labels %s", labels))
 	return services
 }
 
-func CountPodsWithLabel(t *testing.T, namespace, label string) int {
-	pods := getPodsWithLabel(t, namespace, label)
+func CountPodsWithLabels(t *testing.T, namespace string, labels map[string]string) int {
+	pods := getPodsWithLabels(t, namespace, labels)
 	return len(pods.Items)
 }
 
-func PodWithLabelIsReady(t *testing.T, namespace string, label string) bool {
+func PodWithLabelsIsReady(t *testing.T, namespace string, label map[string]string) bool {
 	g(t).Eventually(func() bool {
 		return resourceWithLabelIsPresent(t, namespace, "pod", label)
 	}, retryTimeout, retryInterval).Should(BeTrue())
 
-	pods := getPodsWithLabel(t, namespace, label)
+	pods := getPodsWithLabels(t, namespace, label)
 	if len(pods.Items) == 1 {
 		return strings.ToLower(string(pods.Items[0].Status.Phase)) == "running"
 	}
 	return false
 }
 
-func WaitForPodWithLabelToBeReady(t *testing.T, namespace, label string) {
+func WaitForPodWithLabelsToBeReady(t *testing.T, namespace string, labels map[string]string) {
 	g(t).Eventually(func() bool {
-		return PodWithLabelIsReady(t, namespace, label)
+		return PodWithLabelsIsReady(t, namespace, labels)
 	}, retryTimeout, retryInterval).Should(BeTrue())
 }
 
 func DeployMinioAndCreateBucket(t *testing.T, bucketName string) {
 	helmOptions := &helm.Options{
 		KubectlOptions: getKubectlOptions("default"),
 	}
-	helm.RunHelmCommandAndGetOutputE(t, helmOptions, "repo", "add", "minio", "https://helm.min.io/")
 
-	helm.RunHelmCommandAndGetOutputE(t, helmOptions, "install",
-		"--set", fmt.Sprintf("accessKey=minio_key,secretKey=minio_secret,defaultBucket.enabled=true,defaultBucket.name=%s", bucketName),
-		"minio", "minio/minio", "-n", "minio", "--create-namespace")
+	if _, err := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "repo", "add", "minio", "https://helm.min.io/"); err != nil {
+		t.Fatalf("failed to add minio helm repo: %s", err)
+	}
+
+	values := fmt.Sprintf("accessKey=minio_key,secretKey=minio_secret,defaultBucket.enabled=true,defaultBucket.name=%s", bucketName)
+	if _, err := helm.RunHelmCommandAndGetOutputE(t, helmOptions, "install", "--set", values, "minio", "minio/minio", "-n", "minio", "--create-namespace"); err != nil {
+		t.Fatalf("failed to install the minio helm chart: %s", err)
+	}
```

Comment:
Same question as before with not using Gomega for (async/sync) assertions: would it work for you to use it instead of calling `t.Fatalf()` ?